mm-te: refactor helm chart to deploy MM-TE using the configuration in the DB (#225)
mattermost/mattermost-helm: charts/mattermost-team-edition/README.md

configuration. Note that this failure will occur before any changes have been
made to the k8s cluster.

## Upgrading the Chart to 4.0.0+

The Chart version 4.0.0+ supports only Helm v3, follow the [guide](https://helm.sh/blog/migrate-from-helm-v2-to-helm-v3/)

## Upgrading the Chart to 5.0.0+

> There are some changes that was introduced in Mattermost version 5.30+ that made this helm chart incompatible and the upgrade to fail. To fix that a new Major chart was relased to fix that, and now the configuration is moved to the Database and to environment variables

Reviewer: ```suggestion
There were some changes that was introduced in Mattermost version 5.30+ that made this helm chart incompatible and caused the upgrade to fail. To fix that a new Major version of the chart was released, and now the configuration is moved to the database and environment variables.
```

```

2 - Access the Mattermost pod

```
$ kubectl exec -it mm-te-mattermost-team-edition-799cc8b475-twglt /bin/sh
```

3 - Copy the exist config.json to a temp folder, run this inside the container you accessed on Item 2

```
$ cp config/config.json /tmp
```

4 - Migrate the config.json to the database (run inside the container that you accessed in the Item 2)

```
~ $ ./bin/mattermost config migrate /tmp/config-mm.json "mysql://mmuser:mmuser123@tcp(mm-te-mysql:3306)/mattermost?charset=utf8mb4,utf8&readTimeout=30s&writeTimeout=30s"
{"level":"warn","msg":"DefaultServerLocale must be one of the supported locales. Setting DefaultServerLocale to en as default value."}
{"level":"warn","msg":"DefaultClientLocale must be one of the supported locales. Setting DefaultClientLocale to en as default value."}
{"level":"warn","msg":"DefaultServerLocale must be one of the supported locales. Setting DefaultServerLocale to en as default value."}
{"level":"warn","msg":"DefaultClientLocale must be one of the supported locales. Setting DefaultClientLocale to en as default value."}
{"level":"info","msg":"Successfully migrated config."}
```

5 - Update the helm repo

```
$ helm repo update
```

> 6 - upgrade your mattermost helm, check your existing values, you remove the section `config.json` from your custom values

Reviewer: ```suggestion
6 - Upgrade your mattermost helm, check your existing values, you remove the section `config.json` from your custom values
```

and **important** need to keep the existing image tag in this case was 5.29.0

```
$ helm upgrade  mm-te -f custom_values.yaml --set image.tag=5.29.0 mattermost/mattermost-team-edition
```

7 - It should deploy and in a few seconds the new Pod for the Mattermost server should be up and running

8 - After all up and running you can upgrade it again to get the latest image tag

```
$ helm upgrade  mm-te -f custom_values.yaml mattermost/mattermost-team-edition
```

> if in your `custom_values.yaml` you set the image.tag, please update that to the latest available, at the time of this doc was write the version is `5.35.3`

Reviewer: ```suggestion
If in your `custom_values.yaml` you set the image.tag, please update that to the latest available, at the time of this doc was write the version is `5.35.3`
```
